File: flutter/darwin/sqlite3_flutter_libs/Sources/sqlite3_flutter_libs/Sqlite3FlutterLibsPlugin.swift
import Foundation

#if os(iOS)
import Flutter
#elseif os(macOS)
import FlutterMacOS
#endif

/// Flutter plugin that provides custom SQLite with extensions.
///
/// This plugin itself is a no-op - its purpose is to ensure the custom SQLite
/// library (with FTS5, USearch, SQLean extensions) is linked into the app.
/// The actual SQLite symbols come from the SQLiteDataCustom SPM package.
public class Sqlite3FlutterLibsPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        // No method channel needed on iOS/macOS.
        // The plugin just ensures SQLite is linked via the SPM dependency.
    }
}

File: flutter/android/src/main/kotlin/eu/simonbinder/sqlite3_flutter_libs/Sqlite3FlutterLibsPlugin.kt
package eu.simonbinder.sqlite3_flutter_libs

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * Flutter plugin that bundles custom SQLite with extensions.
 *
 * This plugin ensures the custom SQLite library (with FTS5, USearch, SQLean extensions)
 * is bundled with the app via the sqlite-android dependency. The plugin class itself
 * is a no-op required by Flutter's plugin system.
 */
class Sqlite3FlutterLibsPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
